arch/risc-v/src/common/riscv_macros.S

```diff
@@ -359,7 +359,7 @@
  ****************************************************************************/
 
 .macro  riscv_mhartid out
-#ifdef CONFIG_RISCV_PERCPU_SCRATCH
```

Comment:
This dependency is exactly to the PERCPU_SCRATCH register? If it is not initialized you will return garbage by reading it (you will dereference NULL in fact)